File: learning-dotnet/signalr-terminal/src/types/trading.ts
export interface PricePoint {
  timestamp: number
  price: number
}

export interface StockSnapshot {
  symbol: string
  name: string
  price: number
  open: number
  high: number
  low: number
  change: number
  changePercent: number
  history: PricePoint[]
}

export interface Transaction {
  id: string
  symbol: string
  type: 'BUY' | 'SELL'
  price: number
  quantity: number
  timestamp: number
  trader: string
}

export interface MarketState {
  isOpen: boolean
  stocks: StockSnapshot[]
  recentTransactions: Transaction[]
}

export interface PriceUpdate {
  symbol: string
  price: number
  change: number
  changePercent: number
  timestamp: number
}

File: learning-dotnet/signalr-terminal/Models/Trading.cs
namespace SignalRWebpack.Models;

public record PricePoint(long Timestamp, double Price);

public record StockSnapshot(
    string Symbol,
    string Name,
    double Price,
    double Open,
    double High,
    double Low,
    double Change,
    double ChangePercent,
    List<PricePoint> History);

public record Transaction(
    string Id,
    string Symbol,
    string Type,
    double Price,
    int Quantity,
    long Timestamp,
    string Trader);

public record MarketStateSnapshot(
    bool IsOpen,
    List<StockSnapshot> Stocks,
    List<Transaction> RecentTransactions);

public record TradeRequest(string Symbol, int Quantity);
